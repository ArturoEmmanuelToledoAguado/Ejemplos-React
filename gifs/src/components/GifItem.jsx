export const GifItem = ({title,url}) => {

    console.log(url)

    const copyPath = ()=>{
        navigator.clipboard.writeText(url)
    }

  return (
    <div className="">
        <img 
            src={url}
            alt={title}
        />
        <p>{title}</p>
        <button className=""onClick={copyPath}>Copy Path</button>
    </div>
  )
}
