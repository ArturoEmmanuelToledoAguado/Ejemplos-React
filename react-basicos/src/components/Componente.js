import React from "react";
//import React,{Component} from "react";

const Componente=props=> <h2>{props.msg}</h2>

/* Componente en funcion declarada
function Componente(props) {
    return <h2>{props.msg}</h2>;
}*/

/* Componente con clase
class Componente extends Component{
    render() {
        return <h2>{this.props.msg}</h2>;
    }
}*/

export default Componente;