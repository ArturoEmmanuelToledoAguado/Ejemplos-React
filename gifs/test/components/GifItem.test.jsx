import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Pruebas en el GifItem', () => { 

    const title = 'Evangelion'
    const url= 'https://github.com/'

    test('Esta prueba es para hacer match con el snapshot', () => { 
        const {container} = render(<GifItem title={title} url={url}/>)//Renderiza el componente
        //El snapshot guarda lo que regresa el render
        expect(container).toMatchSnapshot();//Toma foto
     })

    test('Esta prueba es para mostrar la imagen con el URL y el ALT indicado', () => { 
        render(<GifItem title={title} url={url}/>)
        //screen.debug()//Podemos ver todo lo que renderizan (donde estoy parado)
        const{src,alt}=screen.getByRole('img')
        expect(src).toBe(url)
        expect(alt).toBe(title)
      })

    test('Esta prueba es para mostrar el title del componente', () => { 
        render(<GifItem title={title} url={url}/>)
        expect(screen.getByText(title)).toBeTruthy()//Obtenemos el texto que renderiza el componente
       })

    test('Esta prueba valida que el boton existe y pertenece a la clase correcta', () => { 
        render(<GifItem title={title} url={url}/>)
        const {className} = screen.getAllByRole('button')
        expect(screen.getAllByRole('button')).toBeTruthy()
        expect(className).toBe('button')
     })

    test('Esta prueba es para validar la funciÃ³n del boton', () => { 
        const {onClick} = screen.getByRole('button')
        console.log(onClick)
     })
 })