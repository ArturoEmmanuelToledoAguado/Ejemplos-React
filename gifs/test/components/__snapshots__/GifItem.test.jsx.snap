// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el GifItem Esta prueba es para hacer match con el snapshot 1`] = `
<div>
  <div
    class=""
  >
    <img
      alt="Evangelion"
      src="https://github.com/"
    />
    <p>
      Evangelion
    </p>
    <button
      class=""
    >
      Copy Path
    </button>
  </div>
</div>
`;
